systemtest/src/main/java/io/strimzi/systemtest/utils/VerificationUtils.java

```diff
@@ -0,0 +1,463 @@
+                    assertThat(sa.getMetadata().getLabels().get(Labels.STRIMZI_KIND_LABEL), is(Kafka.RESOURCE_KIND));
+                }
+                assertThat(sa.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL), is(clusterName));
+            }
+        );
+    }
+
+    /**
+     * Verifies that the Docker images used by the Kafka cluster in the specified namespaces are correct
+     * based on the configured versions and deployment configurations.
+     *
+     * @param clusterName The name of the Kafka cluster
+     * @param clusterOperatorNamespaceName The namespace where the Cluster Operator is deployed
+     * @param kafkaNamespaceName The namespace where Kafka is deployed
+     * @param kafkaPods The number of Kafka pods in the cluster
+     * @param zkPods The number of ZooKeeper pods in the cluster
+     * @param rackAwareEnabled Indicates whether rack-aware configuration is enabled
+     */
+    public static void verifyKafkaClusterDockerImages(String clusterName, String clusterOperatorNamespaceName, String kafkaNamespaceName, int kafkaPods, int zkPods, boolean rackAwareEnabled) {
+        LOGGER.info("Verifying docker image names");
+        Map<String, String> imgFromDeplConf = getImagesFromConfig(clusterOperatorNamespaceName);
+
+        String kafkaVersion = Crds.kafkaOperation(kubeClient(kafkaNamespaceName).getClient()).inNamespace(kafkaNamespaceName).withName(clusterName).get().getSpec().getKafka().getVersion();
+        if (kafkaVersion == null) {
+            kafkaVersion = Environment.ST_KAFKA_VERSION;
+        }
+
+        if (!Environment.isKRaftModeEnabled()) {
+            //Verifying docker image for zookeeper pods
+            for (int i = 0; i < zkPods; i++) {
+                String imgFromPod = PodUtils.getContainerImageNameFromPod(kafkaNamespaceName, KafkaResources.zookeeperPodName(clusterName, i), "zookeeper");
+                assertThat("ZooKeeper Pod: " + i + " uses wrong image", imgFromPod, containsString(TestUtils.parseImageMap(imgFromDeplConf.get(KAFKA_IMAGE_MAP)).get(kafkaVersion)));
+            }
+        }
+
+        //Verifying docker image for kafka pods
+        for (int i = 0; i < kafkaPods; i++) {
+            String imgFromPod = PodUtils.getContainerImageNameFromPod(kafkaNamespaceName, KafkaResource.getKafkaPodName(clusterName, i), "kafka");
+            assertThat("Kafka Pod: " + i + " uses wrong image", imgFromPod, containsString(TestUtils.parseImageMap(imgFromDeplConf.get(KAFKA_IMAGE_MAP)).get(kafkaVersion)));
+            if (rackAwareEnabled) {
+                String initContainerImage = PodUtils.getInitContainerImageName(KafkaResources.kafkaPodName(clusterName, i));
+                assertThat(initContainerImage, is(imgFromDeplConf.get(KAFKA_INIT_IMAGE)));
+            }
+        }
+
+        //Verifying docker image for entity-operator
+        String entityOperatorPodName = KubeClusterResource.cmdKubeClient(kafkaNamespaceName).listResourcesByLabel("pod",
```

Comment:
I guess you can do it in "more updated way" than this. We have methods for picking label selector atd. so doing this is not that clear FMPOV